std/string.d

```diff
@@ -2638,6 +2638,15 @@ auto lineSplitter(KeepTerminator keepTerm = KeepTerminator.no, Range)(Range r)
     return LineSplitter!(keepTerm, Range)(r);
 }
 
+///
+@safe pure unittest
+{
+    import std.algorithm.comparison : equal;
+
+    string s = "Hello\nmy\rname\nis";
+    assert(splitLines(s).equal(lineSplitter(s)));
```

Comment:
This is fine but maybe a comment on lazyiness of  lineSplitter vs splitLines? e.g. splitLines being an array can be sorted directly.